query/src/test/java/org/infinispan/configuration/AsyncIndexConfigValidationTest.java

```diff
@@ -0,0 +1,81 @@
+package org.infinispan.configuration;
+
+import static org.infinispan.test.TestingUtil.withCacheManager;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.CacheException;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.cache.Index;
+import org.infinispan.hibernate.search.spi.InfinispanIntegration;
+import org.infinispan.query.helper.StaticTestingErrorHandler;
+import org.infinispan.query.indexmanager.InfinispanIndexManager;
+import org.infinispan.query.test.Person;
+import org.infinispan.test.AbstractInfinispanTest;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.testng.annotations.Test;
+
+/**
+ * @since 9.1
+ */
+@Test(groups = "functional", testName = "configuration.AsyncIndexConfigValidationTest")
+public class AsyncIndexConfigValidationTest extends AbstractInfinispanTest {
+
+   @Test(expectedExceptions = CacheException.class)
```

Comment:
actually...Not sure if testng supports matching inner exceptions. The real message is wrapped 3 times unfortunately